Mathlib/LinearAlgebra/Eigenspace/Basic.lean

```diff
@@ -578,3 +578,54 @@ lemma map_add_of_iInf_genEigenspace_ne_bot_of_commute [NoZeroSMulDivisors R M]
 end End
 
 end Module
+
+section SpectrumDiagonal
+
+variable {R : Type*} [Field R] {n : Type*} [DecidableEq n][Fintype n]
+
+open Module.End
+
+/--  Standard basis vectors are eigenvectors of any associated diagonal linear operator. -/
+lemma Matrix.hasEigenvector_toLin'_diagonal (d : n → R) (i : n) :
+    Module.End.HasEigenvector (Matrix.toLin' (diagonal d)) (d i) (Pi.basisFun R n i) := by
```

Comment:
Is this also true for something like
```suggestion
    Module.End.HasEigenvector (Matrix.toLin b b (diagonal d)) (d i) (b i) := by
```
for some `b : Basis _ _ _`? If so, it would be good to have both versions, preferably with one proof using the other.

Reply:
I'll look into this. Thanks!